src/System.Private.CoreLib/shared/System/Number.Formatting.cs

```diff
@@ -387,61 +398,34 @@ public static bool TryFormatDouble(double value, ReadOnlySpan<char> format, Numb
-            int precision = DoublePrecision;
+
             NumberBuffer number = default;
             number.kind = NumberBufferKind.Double;
 
-            switch (fmt)
-            {
-                case 'R':
-                case 'r':
-                    {
-                        // In order to give numbers that are both friendly to display and round-trippable, we parse the
-                        // number using 15 digits and then determine if it round trips to the same value. If it does, we
-                        // convert that NUMBER to a string, otherwise we reparse using 17 digits and display that.
-                        DoubleToNumber(value, DoublePrecision, ref number);
-                        if (number.scale == ScaleNAN)
-                        {
-                            return info.NaNSymbol;
-                        }
-                        else if (number.scale == ScaleINF)
-                        {
-                            return number.sign ? info.NegativeInfinitySymbol : info.PositiveInfinitySymbol;
-                        }
-
-                        if (NumberToDouble(ref number) == value)
-                        {
-                            NumberToString(ref sb, ref number, 'G', DoublePrecision, info);
-                        }
-                        else
-                        {
-                            DoubleToNumber(value, 17, ref number);
-                            NumberToString(ref sb, ref number, 'G', 17, info);
-                        }
+            int precision;
 
-                        return null;
-                    }
-
-                case 'E':
-                case 'e':
-                    // Round values less than E14 to 15 digits
-                    if (digits > 14)
-                    {
-                        precision = 17;
-                    }
-                    break;
+            if ((fmt == 'R') || (fmt == 'r'))
+            {
+                fmt = 'G';
+                precision = DefaultDoublePrecision;
 
-                case 'G':
-                case 'g':
-                    // Round values less than G15 to 15 digits. G16 and G17 will not be touched.
-                    if (digits > 15)
-                    {
-                        precision = 17;
-                    }
-                    break;
+                // IEEE Roundtripping requires us to include at least `17` digits
+                digits = precision;
+            }
+            else if (digits <= 0)
+            {
+                // This ensures that, for the default case, we return a string containing the old
+                // number of digits (15), which results in "prettier" numbers for most cases.
+                precision = DefaultDoubleDigits;
+            }
+            else
+            {
+                // IEEE requires we correctly round to the requested number of digits
+                precision = digits;
```

Comment:
Had to update this logic so that we preserve the fact that the user didn't explicitly specify a digit count, or that the user specified `0` for the digit count.

This makes a difference for things like `double.ToString("N")` which i had accidentally made to be treated like `double.ToString("N15")`.

Reply:
Now, `digits` is always preserved (except for `"R"`, where it is explicitly documented to be ignored) and precision is updated accordingly.

Reply:
This brought the CoreFX failure count down to 354, and they are now all XML and JSON serialization differences plus the UTF8Parser differences (which need fixup and for which the code lives in CoreFX).